dev.py

```diff
@@ -16,63 +16,115 @@
 This is different from a normal boostrapping process because the
 buildout egg itself is installed as a develop egg.
 """
+import sys
 
-import os, shutil, sys, subprocess
+if sys.version_info < (2, 7):
+    raise SystemError("Outside Python 2.7, no support for Python 2.x.")
+
+if sys.version_info > (3, ) and sys.version_info < (3, 5):
+    raise SystemError("No support for Python 3.x under 3.5.")
+
+
+import os, shutil, sys, subprocess, tempfile
 
 for d in 'eggs', 'develop-eggs', 'bin', 'parts':
     if not os.path.exists(d):
         os.mkdir(d)
 
+bin_buildout = os.path.join('bin', 'buildout')
+if os.path.isfile(bin_buildout):
+    os.remove(bin_buildout)
+
 if os.path.isdir('build'):
     shutil.rmtree('build')
 
 ######################################################################
 # Make sure we have a relatively clean environment
-try:
-    import pkg_resources, setuptools
-except ImportError:
-    pass
-else:
-    raise SystemError(
-        "Buildout development with a pre-installed setuptools or "
-        "distribute is not supported.")
 
-######################################################################
-# Install distribute
-ez = {}
+if '--no-clean' not in sys.argv:
+    try:
+        import pip
+        print('Remove pip, setuptools, wheel')
+        print('')
+        if subprocess.call(
+            [sys.executable] + ['-m', 'pip', 'uninstall', '-y', 'setuptools', 'pip',  'wheel']
+        ):
+            raise SystemError(
+                "Buildout development with pre-installed pip and setuptools"
+                "Could not uninstall with pip"
```

Comment:
These two literals get concatenated without any whitespace, resulting in "... and setuptoolsCould not uninstall ...".


```suggestion
                "Buildout development with pre-installed pip and setuptools\n"
                "Could not uninstall with pip"
```